src/pages/ManageStockPage.js

```diff
@@ -0,0 +1,329 @@
+    key: 'temperatureExposure',
+    width: 1,
+    title: LOCALIZATION.headers.temperatureExposure,
+    alignText: 'center',
+  },
+};
+
+const KEY_TO_FUNCTION_MAPPINGS = {
+  totalBatchesInStock: 'getQuantityInLocation',
+  hasBreach: 'getHasBreachedBatches',
+  quantityInBreach: 'getQuantityInBreach',
+  temperatureExposure: 'getTemperatureExposure',
+};
+
+const VACCINE_COLUMN_KEYS = [
+  'code',
+  'name',
+  'totalBatchesInStock',
+  'totalQuantity',
+  'quantityInBreach',
+  'temperatureExposure',
+  'hasBreach',
+  'navigation',
+];
+
+/**
+ * HELPER METHODS
+ */
+const getColumns = () => VACCINE_COLUMN_KEYS.map(key => TABLE_COLUMNS[key]);
+
+/**
+ * Page for managing items. Used for vaccine items currently.
+ * Renders a table, searchable by Item.code or Item.name. Can
+ * apply a location filter showing items for that location and
+ * altering the data within the table to be values for that
+ * location only.
+ * Only props required are acquired through redux.
+ * Optional prop of initialLocation, pre-filtering the list of
+ * items.
+ */
+export class ManageStockPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // List of location objects populated in ComponentDidMount.
+    // Has a special object for 'No Location' to be selected,
+    // which has an id of -1.
+    this.LOCATION_FILTERS = null;
+    // Store for all items. Is used to apply location and searchTerm
+    // filters and derive the next GenericDataTable data prop.
+    this.ITEMS = null;
+
+    this.state = {
+      // Key determining which modal should be rendered. Uses
+      // MODAL_KEYS constant.
+      modalKey: null,
+      isModalOpen: false,
+      // Filter variables: Location object from this.LOCATION_FILTERS.
+      // and searchTerm key from the search bar in GenericDataTable.
+      locationFilter: null,
+      searchTerm: '',
+    };
+  }
+
+  /**
+   * HELPER METHODS
+   */
+
+  locationFilter = (location, item) => {
+    const { id = -1 } = location;
+    if (id === -1) return true;
+    return item.hasBatchInFridge(location);
+  };
+
+  nameAndCodeFilter = (searchTerm, { name, code } = {}) =>
+    name.toLowerCase().startsWith(searchTerm) || code.toLowerCase().startsWith(searchTerm);
+
+  // Called on every render(). Applies the location filter then
+  // searchTerm filter on this.ITEMS.
+  getData = () => {
+    const { locationFilter, searchTerm } = this.state;
+    if (!locationFilter) return [];
+    return this.ITEMS.filter(item => this.locationFilter(locationFilter, item)).filter(item =>
+      this.nameAndCodeFilter(searchTerm, item)
+    );
+  };
+
+  // Gets and returns the Location object set as this.state.locationFilter.
+  // returns undefined if the locationFilter is the 'special' location, AllLocations.
+  getLocation = () => {
+    const { locationFilter } = this.state;
+    const location = !locationFilter || locationFilter.id === -1 ? undefined : locationFilter;
+    return location;
```

Comment:
Need undefined for the location:

The getters use a default value. The default value is only used when no parameter is passed (`undefined`). Passing `null` or some other value will not use the default value. Passing `null` explicitly, will throw an error as `null` is an object and it will try to de structure it. Rather than doing a bunch of logic to either call the the function with the location or not, it will just call with undefined explicitly

Have changed this to account for no -1 : 3913727fdddcb02ce29f5818b3f4653210fd866d